docroot/modules/custom/json_data_field/src/Plugin/Field/FieldType/JsonDataFieldItem.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\json_data_field\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Component\Serialization\Yaml;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Defines the 'json_data' field type.
+ *
+ * @FieldType(
+ *   id = "json_data",
+ *   label = @Translation("JSON (stored as text in database)"),
+ *   description = @Translation("Allows JSON data to be stored in the database. In the database the content is stored in a text column."),
+ *   category = @Translation("Data"),
+ *   default_widget = "yaml_textarea",
+ *   default_formatter = "json_data_formatter",
+ * )
+ */
+class JsonDataFieldItem extends FieldItemBase {
+
+  use StringTranslationTrait;
+
+  /**
+   * Json data medium text 16MB/utf8mb4.
+   */
+  const SIZE = 4194304;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultStorageSettings() {
+    return [
+      'size' => static::SIZE,
+    ] + parent::defaultStorageSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    $properties['value'] = DataDefinition::create('string')
+      ->setLabel(new TranslatableMarkup('YAML Value'))
+      ->setDescription(new TranslatableMarkup('Settings value for this instance'));
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    $schema['columns']['value'] = [
+      'description' => 'Json data.',
+      'type' => 'text',
+      'size' => 'medium',
+      'serialize' => TRUE,
+    ];
+    return $schema;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setValue($values, $notify = TRUE) {
+    // Make sure the values should be an array.
+    if (isset($values['value']) && is_array($values)) {
+
+      // Decode the YAML data into array.
+      $yaml_array = Yaml::decode($values['value']);
+
+      // Convert the yaml array into JSON data.
+      $json_data = json_encode($yaml_array, TRUE);
```

Comment:
This should not be necessary if `Serialized` is true in the schema. The whole point of using Serialized is that we can store PHP associative arrays in the DB without having to make JSON

Reply:
Thanks Bryan. I removed this code from field item and added in widget level..